src/main/java/com/google/sps/servlets/UserDataServlet.java

```diff
@@ -1,59 +1,57 @@
 package com.google.sps.servlets;
 
 import com.google.sps.data.UserData;
+import com.google.sps.data.FirebaseSingletonApp;
 import com.google.gson.Gson;
-import java.util.List;
 import java.io.IOException;
-import java.util.ArrayList;
 import com.google.firebase.FirebaseApp;
-import com.google.firebase.FirebaseOptions;
 import com.google.firebase.auth.*;
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
-import com.google.appengine.api.datastore.Query.SortDirection;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
 /**
- * Servlet that will add a user with his address to the database when he sets it first.
- * Users can also edit their adrress. 
+ * Servlet that will add a user with his address to the database when he sets it
+ * first. Users can also edit their adrress.
  */
 @WebServlet("/user-data")
 public class UserDataServlet extends HttpServlet {
-  
+
   // Get all info about current user.
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-  
+
     // Initialize firebase app.
-    FirebaseApp defaultApp = initializeFirebaseApp(/* setOptions */ false);
+    FirebaseApp firebaseApp = FirebaseSingletonApp.getInstance();
     String uid = null;
     try {
-      // get the idToken from hidden input field
+      // get the idToken from query string
       String idToken = request.getParameter("idToken");
```

Comment:
I'm not sure if it's a problem with sending the token ID (a private information of the user) in a GET request, but I found that in the documentation, it's recommended to send it via POST: 
https://developers.google.com/identity/sign-in/android/backend-auth